### ISSUE TITLE
Feature Request: Systems Manager Patch Baseline Data Source 

### ISSUE THREAD

**Issue by ericwestfall** [terraform-providers/terraform-provider-aws#5116]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

A patch baseline defines which patches are approved for installation on your instances. This data source will provide a mechanism to search for and retrieve details about a patch baseline. Search filters to include `OS`, `Owner` and a boolean flag to indicate whether the query should return default patch baselines or not. 

### New or Affected Resource(s)

* `data.aws_ssm_patch_baseline`

### Potential Terraform Configuration

```hcl
data "aws_ssm_patch_baseline" "baseline" {
  owner = "aws"
}
```

### References

* #3342

**Comment by jfharden** [comment]
I had a look at adding this data source and it seems the api for searching for patch baselines (DescribePatchBaselines) isn't so friendly. It only allows searching on NAME_PREFIX and OWNERS, but names of patch baselines are not unique, I think this will mean a similar approach to how ami lookups are done with filtering inside the provider on top of the limited filtering provided by the api.

**Comment by jdheyburn** [comment]
Did this develop any further @jfharden ? I wanted to take a look at implementing this so looked to see what difficulties you came across.

An additional use case for this would be to retrieve back the default baselines provided by AWS in order to apply them to a patch group, etc.

**Comment by bflad** [comment]
Support for a new `aws_ssm_patch_baseline` data source has been merged and will release with version 2.48.0 of the Terraform AWS Provider, later next week. Thanks to @jdheyburn for the implementation. 👍
